src/content/docs/en/guides/cms/builderio.mdx

```diff
@@ -3,19 +3,321 @@ title: Builder.io & Astro
+    ></builder-component>
+    <script async src="https://cdn.builder.io/js/webcomponents"></script>
+
+    <footer>This is your footer</footer>
+  </body>
+</html>
+
+```
+
+In the above example, `<builder-component>` tells Builder where to insert the content from its CMS.
+
+#### Setting the new route as the preview URL
+
+1. Copy the full URL of your preview, including the protocol, to your clipboard (e.g. `https://{your host}/builder-preview`). 
+
+2. Go to the **Models** tab in your Builder space, pick the model you've created and paste the URL from step 1 into the **Preview URL** field. Make sure the URL is complete and includes the protocol, for example `https://`.
+
+3. Click the **Save** button in the upper right.
+
+:::tip
+When you deploy your site, change the preview URL to match your production URL, for example `https://myAwesomeAstroBlog.com/builder-preview`.
+:::
+
+#### Testing the preview URL setup
+
+1. Make sure your site is live (e.g. your dev server is running) and the `/builder-preview` route is working.
+
+2. In your Builder space under the **Content** tab, click on **New** to create a new content entry for your `blogpost` model.
+
+3. In the Builder editor that just opened, you should be able to see the `builder-preview.astro` page with a big **Add Block** in the middle.
+
+:::tip[Troubleshooting]
+
+Things can sometimes go wrong when setting up the preview. If something's not right, you can try one of these things:
+
+* Make sure the site is live - for example, your dev server is running.
+* Make sure that the URLs match exactly - the one in your Astro project and the one set in the Builder app.
+* Make sure it's the full URL including the protocol, for example `https://`.
+* If you're working in a virtual environment like [StackBlitz](https://stackblitz.com/) or [Gitpod](https://www.gitpod.io/), you might have to copy and paste the URL again when you restart your workspace, since this usually generates a new URL for your project.
+
+For more ideas, read [Builder's troubleshooting guide](https://www.builder.io/c/docs/guides/preview-url-working).
+:::
+
+### Creating a blog post
+
+1. In Builder's visual editor, create a new content entry with the following values:
+    - **title:** 'First post, woohoo!'
+    - **slug:** 'first-post-woohoo'
+2. Complete your post using the **Add Block** button and add a text field with some post content .
+3. In the text field above the editor, give your entry a name. This is how it will be listed in the Builder app.
+4. When you're ready click the **Publish** button in the upper right corner.
+5. Create as many posts as you like, ensuring that all content entries contain a `title` and a `slug` as well as some post content.
+
+### Displaying a list of blog posts
+
+Add the following content to `src/pages/index.astro` in order to display a list of all post titles, each linking to its own page:
+
+
+
+```astro title="src/pages/index.astro" ins={18-99}
```

Comment:
``````suggestion
Add the following content to `src/pages/index.astro` in order to fetch and display a list of all post titles, each linking to its own page:



```astro title="src/pages/index.astro"
``````